omod/src/test/java/org/openmrs/module/idgen/web/search/LogEntrySearchHandlerTest.java

```diff
@@ -0,0 +1,90 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.idgen.web.search;
+
+import org.junit.Test;
+import org.junit.Before;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import org.junit.Assert;
+
+public class LogEntrySearchHandlerTest extends MainResourceControllerTest {
+    private IdentifierSourceService service;
+
+    @Before
+    public void setUp() throws Exception {
+        executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+        this.service = Context.getService(IdentifierSourceService.class);
+    }
+
+    @Override
+    public long getAllCount() {
+        return service.getLogEntries(null, null, null, null, null, null).size();
+    }
+
+    @Override
+    public String getURI() {
+        return "idgen/logentry";
+    }
+
+    @Override
+    public String getUuid() {
+        return "100892";
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryBySource() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("source", "0d47284f-9e9b-4a81-a88b-8bb42bc0a901");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertNotNull(result);
+        Assert.assertEquals(3, Util.getResultsSize(result));
+
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryByComment() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("comment", "New");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(2, Util.getResultsSize(result));
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryByIdentifier() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("identifier", "H9");
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertEquals(2, Util.getResultsSize(result));
+
+    }
+
+    @Test
+    public void getSearchConfig_shouldReturnLogEntryByDateGenerated() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("fromDate", "2016-10-01 12:00:00.433");
```

Comment:
declare those dates as constants then reference them

Reply:
For this, just like the uuid / identifier above, I am just using then once  within the file. Wouldnt that be adding extra lines of code that could be avoided?

Reply:
That is fine if you are using them once.